src/NuGet.Core/Microsoft.Build.NuGetSdkResolver/RestoreRunnerEx.cs

```diff
@@ -80,10 +91,10 @@ public static Task<IReadOnlyList<RestoreResultPair>> RunWithoutCommit(LibraryIde
                         ProjectStyle = ProjectStyle.PackageReference,
                         ProjectUniqueName = projectFullPath,
                         OutputPath = projectDirectory,
-                        OriginalTargetFrameworks = TargetFrameworks.Select(i => i.ToString()).ToList(),
+                        OriginalTargetFrameworks = otf,
                         ConfigFilePaths = settings.GetConfigFilePaths(),
                         PackagesPath = SettingsUtility.GetGlobalPackagesFolder(settings),
-                        Sources = SettingsUtility.GetEnabledSources(settings).ToList(),
+                        Sources = SettingsUtility.GetEnabledSources(settings).AsList(),
                         FallbackFolders = SettingsUtility.GetFallbackPackageFolders(settings).ToList()
```

Comment:
How is this different than the change included in this PR for GetEnabledSources? Also, I think you might be point out an issue in the implementation of GetFallbackPackageFolders because it's returning a List object without calling AsReadOnly() first. I haven't used IReadOnlyList before so I could be off here.

Reply:
AsReadOnly returns a collection. 

Both methods should probably be `AsList()`

Reply:
Would you please take a look at my second iteration? GetFallbackPackageFolders now correctly returns a read-only collection. Therefore, calling AsList() on this collection won't work. Another option is to change the GetFallbackPackageFolders API so that it returns an object implementing IList.